include/irr/video/IGPUVirtualTexture.h

```diff
@@ -0,0 +1,188 @@
+        uint32_t _tilePadding = 9u,
+        uint32_t _maxAllocatableTexSz_log2 = 14u
+    ) :
+        base_t(
+            _pgTabSzxy_log2,
+            _pgTabLayers,
+            _pgSzxy_log2,
+            _tilePadding
+        ),
+        m_driver(_driver)
+    {
+        m_pageTable = createPageTable(_pgTabSzxy_log2, _pgTabLayers, _pgSzxy_log2, _maxAllocatableTexSz_log2);
+        initResidentStorage(_residentStorageParams, _residentStorageCount);
+    }
+    IGPUVirtualTexture(IVideoDriver* _driver, asset::IAssetManager* _am, asset::ICPUVirtualTexture* _cpuvt) :
+        base_t(
+            _cpuvt->getPageExtent_log2(),
+            _cpuvt->getPageTable()->getCreationParameters().arrayLayers,
+            _cpuvt->getPageExtent_log2(),
+            _cpuvt->getTilePadding(),
+            false
+            ),
+        m_driver(_driver)
+    {
+        //now copy from CPU counterpart resources that can be shared (i.e. just copy state) between CPU and GPU
+        //and convert to GPU those which can't be "shared": page table and VT resident storages along with their images and views
+
+        m_pageTable = std::move(_driver->getGPUObjectsFromAssets(&_cpuvt->getPageTable(), &_cpuvt->getPageTable()+1)->front());
+
+        m_freePageTableLayerIDs = _cpuvt->getFreePageTableLayersStack();
+
+        memcpy(m_layerToViewIndexMapping->data(), _cpuvt->getLayerToViewIndexMapping().begin(), 
+            _cpuvt->getLayerToViewIndexMapping().size()*sizeof(decltype(m_layerToViewIndexMapping)::pointee::value_type));
+
+        const auto& cpuStorages = _cpuvt->getResidentStorages();
+        for (const auto& pair : cpuStorages)
+        {
+            auto* cpuStorage = static_cast<asset::ICPUVirtualTexture::ICPUVTResidentStorage*>(pair.second.get());
+            const asset::E_FORMAT_CLASS fmtClass = pair.first;
+
+            m_storage.insert({fmtClass, core::make_smart_refctd_ptr<IGPUVTResidentStorage>(_driver, _am, cpuStorage)});
+        }
+
+        m_fsamplers.views = _cpuvt->getFloatViews().size() ? _driver->getGPUObjectsFromAssets(_cpuvt->getFloatViews().begin(), _cpuvt->getFloatViews().end()) : nullptr;
+        m_isamplers.views = _cpuvt->getIntViews().size() ? _driver->getGPUObjectsFromAssets(_cpuvt->getIntViews().begin(), _cpuvt->getIntViews().end()) : nullptr;
+        m_usamplers.views = _cpuvt->getUintViews().size() ? _driver->getGPUObjectsFromAssets(_cpuvt->getUintViews().begin(), _cpuvt->getUintViews().end()) : nullptr;
+    }
+
+    STextureData pack(const image_t* _img, const asset::IImage::SSubresourceRange& _subres, asset::ISampler::E_TEXTURE_CLAMP _wrapu, asset::ISampler::E_TEXTURE_CLAMP _wrapv, asset::ISampler::E_TEXTURE_BORDER_COLOR _borderColor) override
+    {
+        assert(0);
+        return STextureData::invalid();
+    }
+
+    bool free(const STextureData& _addr, const asset::IImage* _img, const asset::IImage::SSubresourceRange& _subres) override
+    {
+        assert(0);
+        return false;
+    }
```

Comment:
CPU and GPU specializations don't need to do anything extra for the free than the base